realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -594,6 +594,22 @@ public double average(String fieldName) {
         return where().distinctAsync(fieldName);
     }
 
+    /**
+     * Returns a distinct set of objects from a specific class that fulfill the multiple fields
+     * condition. When multiple distinct fields are given, all unique combinations of tuples in such
+     * fields will be returned. If the result is sorted, the first object of a column will be returned
+     * in case of multiple occurrences, otherwise it is undefined which object is returned.
+     *
+     * @param firstFieldName first field name to use when finding distinct objects.
+     * @param remainingFieldNames remaining field names when determining all unique combinations of field values.
+     * @return a non-null {@link RealmResults} containing the distinct objects.
+     * @throws IllegalArgumentException if fieldNames is empty, or a field is null, does not exist,
```

Comment:
It should still work if the field have a null? I assume you mean the field name is null? Perhaps `if a field name is empty or {@code null}, does not exist, is an unsupported type or points to a linked field.